cmd/blobstream/query/cmd.go

```diff
@@ -478,3 +497,82 @@ func getSignatureAndPrintIt(
 	}
 	return nil
 }
+
+// tryToGetExistingConfig tries to get the query config from existing
+// orchestrator/relayer homes. It first checks whether the `--home` flag was
+// changed. If so, it gets the config from there. If not, then it tries the
+// orchestrator default home directory, then the relayer default home directory.
+func tryToGetExistingConfig(cmd *cobra.Command, logger tmlog.Logger) (Config, error) {
+	v := viper.New()
+	v.SetEnvPrefix("")
+	v.AutomaticEnv()
+	homeDir, changed, err := base.GetHomeFlag(cmd)
+	if err != nil {
+		return Config{}, err
+	}
+	// the --home flag was set to some directory
+	if changed && homeDir != "" {
+		logger.Debug("using home", "home", homeDir)
+		configPath := filepath.Join(homeDir, "config")
+
+		// assume this home is an orchestrator home directory
+		orchConf, err := orchestrator.GetStartConfig(v, configPath)
+		if err == nil {
+			// it is an orchestrator, so we get the config from it
+			return Config{
+				coreGRPC:     orchConf.CoreGRPC,
+				coreRPC:      orchConf.CoreRPC,
+				targetNode:   orchConf.Bootstrappers,
+				grpcInsecure: orchConf.GRPCInsecure,
+			}, nil
+		}
+
+		// assume this home is a relayer home directory
+		relConf, err := relayer.GetStartConfig(v, configPath)
+		if err == nil {
+			// it is a relayer, so we get the config from it
+			return Config{
+				coreGRPC:     relConf.CoreGRPC,
+				coreRPC:      relConf.CoreRPC,
+				targetNode:   relConf.Bootstrappers,
+				grpcInsecure: relConf.GrpcInsecure,
+			}, nil
+		}
+		return Config{}, fmt.Errorf("the provided home directory is neither an orchestrator nor a relayer home directory")
+	}
+	// try to get the config from the orchestrator home directory
+	orchHome, err := base.GetHomeDirectory(cmd, orchestrator.ServiceNameOrchestrator)
+	if err != nil {
+		return Config{}, err
+	}
+	orchConf, err := orchestrator.GetStartConfig(v, filepath.Join(orchHome, "config"))
+	if err == nil {
+		// found orchestrator home, get the config from it
+		logger.Debug("using home", "home", orchHome)
+		return Config{
+			coreGRPC:     orchConf.CoreGRPC,
+			coreRPC:      orchConf.CoreRPC,
+			targetNode:   orchConf.Bootstrappers,
+			grpcInsecure: orchConf.GRPCInsecure,
+		}, nil
```

Comment:
[nit]

since we're repeating this logic so much, imo we should create a method to create the config and return it

Reply:
[408de1d](https://github.com/celestiaorg/orchestrator-relayer/pull/619/commits/408de1d2b299b7524d8153658b8d6d4d7564744c)

Reply:
not blocking ofc, but what I meant here was something along the lines of

```go
orchConfig.Config()
```

just something to eliminate having to have the repeat having to reference the same fields over and over again

Reply:
those are two different structs from different contexts. I don't think it's better to have such method. One is orchestrator configuration and the other is a query config. The reason I am using it is to avoid duplicating the orchestrator's file logic